Improve wallet-api

Happy2C0de opened the pull request:
- Change POST to DELETE on credential endpoint

Happy2C0de commented:
This makes just no sense.

mikeplotean commented:
then we can revert to using the delete verb once the update to caddy in docker-compose is ready

Happy2C0de commented:
@waltkb @mikeplotean 
The problem isn't on nginx but on ktor CORS. See the changes. 
Can you try the setup on your side?

mikeplotean commented:
> @waltkb @mikeplotean The problem isn't on nginx but on ktor CORS. See the changes. Can you try the setup on your side?

this won't help, because the ktor cors plugin being used, supports these by default, please see https://ktor.io/docs/cors.html#methods
this is an nginx related issue

Happy2C0de commented:
> > @waltkb @mikeplotean The problem isn't on nginx but on ktor CORS. See the changes. Can you try the setup on your side?
> 
> unfortunately, this won't help, please see the link mentioned above this is an nginx related issue

The link [CORS - Methods](https://ktor.io/docs/cors.html#methods=) mentions 
> By default, the CORS plugin allows the GET, POST and HEAD HTTP methods. 

`DELETE` is not supported by default. For me, the provided PR fixed the local docker-compose setup and I can make delete requests.

mikeplotean commented:
yes, my mistake, you probably saw the original message
don't know how the provided pr was tested, but the record doesn't get deleted from the db

Happy2C0de commented:
> Hi @Happy2C0de, I'll close this PR, I've fixed this issue in above mentioned PR #122, and updated the docker ingress to also be able to utilize proper HTTP operations.

That's fine. You mitigated the problem by switching to Caddy. Thanks.
